### ISSUE TITLE
Consolidate weighted and unweighted versions of eccentricity-related functions

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2683]
Eccentricity-related functions have unweighted versions, such as `igraph_eccentricity()`, and weighted versions, such as `igraph_eccentricity_dijkstra()`.

I propose keeping the unweighted name only, and passing `weights=NULL` to request unweighted calculations. Duplicating names is overkill. We do this with shortest path functions, for which we have several algorithms implemented, mainly to support negative weights. I would argue that with eccentricity-related concepts, negative weights are not particularly useful anyway.

Let's make the change for these functions:

`igraph_eccentricity`, `igraph_radius`, `igraph_pseudo_diameter`, `igraph_graph_center`, `igraph_diameter`.

Opinions, @ntamas?  This is for 1.0. I can do the refactoring.

**Comment by szhorvat** [comment]
@GroteGnoom , @ntamas : any opinions on this? Any objections?

**Comment by ntamas** [comment]
No objections, feel free to go ahead.

**Comment by szhorvat** [comment]
> How about `igraph_average_path_length()` and `igraph_average_path_length_dijkstra()`?

Yes, absolutely. I would do everything. Ideally I would even do `igraph_distances()`, and add an extra parameter for selecting the method (Dijkstra, Bellman-Ford, etc.), with an "automatic" method that chooses based on the situation.

I'm travelling now and will take some time to respond to everything.

**Comment by ntamas** [comment]
For the record: automatic method selection is covered by https://github.com/igraph/igraph/issues/2481
